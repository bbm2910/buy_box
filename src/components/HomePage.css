.homepage {
	background-image: url('../assets/bg-img.jpeg');
	background-repeat: no-repeat;
	background-size: cover;
	background-position: center;
	height: 100vh;
	border: 1px solid transparent;
	overflow: hidden;
}

.homepage h2 {
	text-transform: uppercase;
	margin: 1%;
	font-size: 2.5rem;
	font-weight: 700;
	color: var(--color1);
	text-align: right;
	white-space: nowrap;
	overflow: hidden;
	transform: translateX(100%);
}

.homepage #text1 {
	animation: slide-left 20s linear infinite 0s;
}

.homepage #text2 {
	animation: slide-right 20s linear infinite 5s;
}

.homepage #text3 {
	animation: slide-left 20s linear infinite 10s;
}

.homepage #text4 {
	animation: slide-right 20s linear infinite 15s;
}

.homepage #text5 {
	animation: slide-left 20s linear infinite 20s;
}
.homepage #text6 {
	animation: slide-right 20s linear infinite 20s;
}
.homepage #text7 {
	animation: slide-left 20s linear infinite 20s;
}

@keyframes slide-left {
	0% {
		transform: translateX(40%);
	}
	100% {
		transform: translateX(-100%);
	}
}

@keyframes slide-right {
	0% {
		transform: translateX(-100%);
	}
	100% {
		transform: translateX(100%);
	}
}
