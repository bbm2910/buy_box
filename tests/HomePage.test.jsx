import { render } from '@testing-library/react';
import HomePage from '../src/components/HomePage';

describe('HomePage component', () => {
	it('renders the component with .homepage class', () => {
		const { container } = render(<HomePage />);
		expect(container).toMatchSnapshot();

		const homepageElement = container.querySelector('.homepage');
		expect(homepageElement).toBeInTheDocument();
		expect(homepageElement).toHaveClass('homepage');
	});
});
