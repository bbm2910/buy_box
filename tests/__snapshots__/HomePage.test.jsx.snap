// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage component > renders the component with .homepage class 1`] = `
<div>
  <div
    class="homepage"
  >
    <h2
      id="text1"
    >
      Unbeatable Prices
    </h2>
    <h2
      id="text2"
    >
      Doorbuster Price
    </h2>
    <h2
      id="text3"
    >
      Exclusive Promotions
    </h2>
    <h2
      id="text4"
    >
      Markdown Madness
    </h2>
    <h2
      id="text5"
    >
      Clearance Event
    </h2>
    <h2
      id="text6"
    >
      Limited-Time Offer
    </h2>
    <h2
      id="text7"
    >
      Mega Salet
    </h2>
  </div>
</div>
`;
